pyccel/ast/operators.py

```diff
@@ -761,14 +813,12 @@ class PyccelBooleanOperator(PyccelOperator):
     arg2: PyccelAstNode
         The second argument passed to the operator
     """
-    _dtype = NativeBool()
-    _rank  = 0
-    _shape = ()
-    _precision = default_precision['bool']
 
-    def _set_dtype(self):
+    @staticmethod
+    def _calculate_dtype():
         pass
```

Comment:
Previously we had:
```
def _set_dtype(self):
    pass
```
which means that the dtype and precision were not set. This is done because `self._dtype` already returns something `NativeBool()`. This works because the `_dtype` is a class property. If it is no longer a class property then the dtype needs to be set. So here we should have:
```
@staticmethod
    def _calculate_dtype():
         _dtype = NativeBool()
         _precision = default_precision['bool']
        return _dtype, _precision
```

Reply:
Ok. This was one of the things I didn't understand. Thank you so much.